libparsec/crates/platform_mountpoint/src/unix/filesystem.rs

```diff
@@ -120,25 +143,57 @@ fn entry_stat_to_file_attr(stat: EntryStat, inode: Inode, uid: u32, gid: u32) ->
     }
 }
 
+enum MaybeReadOnly {
+    ReadOnlyMountpoint,
+    ReadWriteMountpoint,
+}
+
+impl MaybeReadOnly {
+    fn get(&self, ops: &WorkspaceOps) -> bool {
```

Comment:
It is not needed to check the current role since access control is done when calling the workspace ops methods, instead we can just rely on the `is_read_only` computed once at mountpoint start, and return the correct error when workspace ops return a not allowed error code

Reply:
28a6bdf858bce7419dfadb38c812d9f486d02aa7